modules/integration/src/test/scala/scala/cli/integration/ReplTestDefinitions.scala

```diff
@@ -51,19 +60,20 @@ abstract class ReplTestDefinitions(val scalaVersionOpt: Option[String])
         }
         .flatMap(arg => Seq("--ammonite-arg", arg))
       val res =
-        os.proc(TestUtil.cli, "--power", "repl", extraOptions, "--ammonite", ammArgs).call(cwd =
-          root
-        )
+        os.proc(TestUtil.cli, "--power", "repl", testExtraOptions, "--ammonite", ammArgs)
+          .call(cwd = root, stderr = os.Pipe)
       val output = res.out.trim()
       expect(output == s"Hello from Scala $expectedAmmoniteVersion")
+      if (useMaxAmmoniteScalaVersion) {
+        // the maximum Scala version supported by ammonite is being used, so we shouldn't downgrade
+        val errOutput = res.err.trim()
+        expect(!errOutput.contains("not yet supported with this version of Ammonite"))
+      }
     }
   }
 
-  test("ammonite") {
-    ammoniteTest()
-  }
-
-  test("ammonite scalapy") {
+  def ammoniteScalapyTest(useMaxAmmoniteScalaVersion: Boolean): Unit = {
```

Comment:
Why do you have this `useMaxAmmoniteScalaVersion` flag? Is there a use case where you want to set it to `false`?

Reply:
Yes. Check `ReplTestsDefault`.
When the default Scala version for users is higher than the max supported for ammonite, we still want to run the test to make sure it falls back to the max gracefully, rather than crashing.

Reply:
Oh, I see what you mean. Somehow I missed that, thanks.